### ISSUE TITLE
asusctl: Unknown interface 'org.asuslinux.Platform, possibly mismatching version of asusd service

### ISSUE THREAD

**Issue by Volotomite** [NixOS/nixpkgs#316538]
### Describe the bug
After installing asusctl in systemPackages with unstable (v6.0.9) and enabling asusd service on my ROG Zephyrus G16 GU605MI, following error occurs upon running asusctl:
```
thread 'main' panicked at asusctl/src/main.rs:61:54:
called `Result::unwrap()` on an `Err` value: MethodError(OwnedErrorName("org.freedesktop.DBus.Error.UnknownInterface"),
Some("Unknown interface 'org.asuslinux.Platform'"), 
Msg { type: Error, serial: 9, sender: UniqueName(":1.159"), reply-serial: 5, body: Signature("s"), fds: [] })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Aborted (core dumped)
```
### Steps To Reproduce
1. Add systemctl unstable to systemPackages
2. enable services.asusd.enable, services.asusd.enableUserService
3. rebuild, reboot - start asusd
4. run asusctl - error occurs

### Expected behavior
asusctl working properly

### Additional context
If I'm not mistaken, it seems to me it has to deal with asusd service version. When I run "systemctl status asusd" I get following output:
```
asusd.service - ASUS Notebook Control
     Loaded: loaded (/etc/systemd/system/asusd.service; linked; preset: enable>
     Active: active (running) since Sat 2024-06-01 19:16:34 EDT; 4min 27s ago
    Process: 7952 ExecStartPre=/nix/store/pxf98n8dsxb6kmm0pdvc3dr3gzcqhizi-cor>
   Main PID: 7954 (asusd)
         IP: 0B in, 0B out
         IO: 13.3M read, 8.0K written
      Tasks: 28 (limit: 37993)
     Memory: 10.4M
        CPU: 229ms
     CGroup: /system.slice/asusd.service
             └─7954 /nix/store/chagnplicbmiz40dipfcv35crbj256y8-asusctl-4.7.1/>

```
In control group it relies on asusctl-4.7.1 - which seems incorrect and very old version.
When I check version of asusctl and asusd package that comes with it I get:
```
 ls -l $(which asusctl)
lrwxrwxrwx 1 root root 69 Dec 31  1969 /run/current-system/sw/bin/asusctl -> /nix/store/j9xl40s15q7lhx1kv8yd2f77b9j20046-asusctl-6.0.9/bin/asusctl

```
```
ls -l $(which asusd)
lrwxrwxrwx 1 root root 67 Dec 31  1969 /run/current-system/sw/bin/asusd -> /nix/store/j9xl40s15q7lhx1kv8yd2f77b9j20046-asusctl-6.0.9/bin/asusd
```

Which shows v6.0.9 and looks correct.
Without enabling asusd service asusctl won't work, so I tried to fix it multiple times by tweaking asusd service and failed (most likely due to lack of experience), trying stuff like this:
https://nixos.org/manual/nixos/unstable/#sec-replace-modules

Overall, this version mismatch behavior seems wrong unless I'm not aware of something. Is there something I can do to resolve this? Thanks.

Is there something I can do to fix it? Thanks in advance.
### Notify maintainers
@K900 @aacebedo

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.9.3, NixOS, 23.11 (Tapir), 23.11.7454.25cf937a30bf`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos-23.11, nixpkgs-unstable, unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by K900** [comment]
You need to set `services.asusd.package = pkgsUnstable.asusctl` or something like that.

**Comment by Volotomite** [comment]
I got this issue resolved - apparently channel version 24.05 got released just after a day I installed nixos and I was using asusd from 23.11 channel which didn't had a "package" attribute which got me really confused.

If by chance someone facing same issue I described above - make sure you're not using channel v23.11

**Comment by yashas3013** [comment]
I am getting the same error running asusctl, but there is no version mismatch

```
thread 'main' panicked at asusctl/src/main.rs:61:54:
called `Result::unwrap()` on an `Err` value: MethodError(OwnedErrorName("org.freedesktop.DBus.Error.UnknownInterface"), Some("Unknown interface 'org.asuslinux.Platform'"), Msg { type: Error, serial: 1222, sender: UniqueName(":1.15"), reply-serial: 5, body: Signature("s"), fds: [] })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Aborted (core dumped)
```

systemctl status asusctl
```
● asusd.service - ASUS Notebook Control
     Loaded: loaded (/etc/systemd/system/asusd.service; linked; preset: enabled)
     Active: active (running) since Thu 2024-12-26 20:37:31 IST; 12min ago
    Process: 1136 ExecStartPre=/nix/store/php4qidg2bxzmm79vpri025bqi0fa889-coreutils-9.5/bin/sleep 1 (code=exited, status=0/SUCCESS)
   Main PID: 1392 (asusd)
         IP: 0B in, 0B out
         IO: 9.1M read, 16.0K written
      Tasks: 22 (limit: 18306)
     Memory: 12.8M (peak: 16.6M)
        CPU: 389ms
     CGroup: /system.slice/asusd.service
             └─1392 /nix/store/j9xl40s15q7lhx1kv8yd2f77b9j20046-asusctl-6.0.9/bin/asusd

Dec 26 20:37:31 nixos asusd[1392]: [WARN  rog_aura::aura_detection] the aura_support.ron file has no entry for this model: GA403UV, 19b6
Dec 26 20:37:31 nixos asusd[1392]: [WARN  rog_aura::aura_detection] the aura_support.ron file has no entry for this model: GA403UV, 19b6
Dec 26 20:37:31 nixos asusd[1392]: [WARN  rog_aura::aura_detection] the aura_support.ron file has no entry for this model: GA403UV, 19b6
Dec 26 20:37:31 nixos asusd[1392]: [WARN  rog_aura::aura_detection] the aura_support.ron file has no entry for this model: GA403UV, 19b6
Dec 26 20:37:31 nixos asusd[1392]: [INFO  rog_aura::aura_detection] Matched to GA403U
Dec 26 20:37:31 nixos asusd[1392]: [INFO  config_traits] Parsed RON for "asusd::ctrl_aura::config::AuraConfig"
Dec 26 20:37:31 nixos asusd[1392]: [INFO  dmi_id] Found dmi ID info at "id"
Dec 26 20:37:31 nixos asusd[1392]: [WARN  asusd::ctrl_aura::controller] No asus::kbd_backlight found for ROG Zephyrus G14 ??
Dec 26 20:37:31 nixos asusd[1392]: [INFO  asusd::ctrl_aura::controller] Found 1 Aura devices
Dec 26 20:37:31 nixos systemd[1]: Started ASUS Notebook Control.
```

version of asusctl and asusd
```
$ ls -l $(which asusctl)
lrwxrwxrwx 1 root root 69 Jan  1  1970 /run/current-system/sw/bin/asusctl -> /nix/store/j9xl40s15q7lhx1kv8yd2f77b9j20046-asusctl-6.0.9/bin/asusctl
```

```
[yas@nixos:~]$ ls -l $(which asusd)
lrwxrwxrwx 1 root root 67 Jan  1  1970 /run/current-system/sw/bin/asusd -> /nix/store/j9xl40s15q7lhx1kv8yd2f77b9j20046-asusctl-6.0.9/bin/asusd
```

I added asusctl by following this (link)[https://asus-linux.org/guides/nixos/]

I tried it 24.11 version NixOS, I am getting the same error which uses 6.0.12 asusctl 
nix-info output:
```
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.32, NixOS, 24.05 (Uakari), 24.05.984.0b8e7a1ae5a9`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

Is there anything that I am missing here?
